components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1345,8 +1349,9 @@ _to_int() {
 # ```
 _port_is_valid() {
     local port="$1"
-    local -i port_num=$(_to_int $port 2>/dev/null)
-    if (( $port_num < 1 || $port_num > 65535 )) ; then
```

Comment:
I've seen this in a few places. What's the reasoning behind splitting the declaration of a local variable and its assignment?

Reply:
Basically, if you run a separate command as part of the assignment, you mask its return value. See https://github.com/koalaman/shellcheck/wiki/SC2155#declare-and-assign-separately-to-avoid-masking-return-values

Reply:
Ah, ok.  TIL